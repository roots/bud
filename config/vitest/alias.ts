import {path} from '@repo/constants'

export default {
  '@roots/bud': path(`sources/@roots/bud/src`),
  '@roots/bud-api': path(`sources/@roots/bud-api/src`),
  '@roots/bud-api/methods': path(`sources/@roots/bud-api/src/methods`),
  '@roots/bud-api/service': path(`sources/@roots/bud-api/src/service`),
  '@roots/bud-babel': path(`sources/@roots/bud-babel/src`),
  '@roots/bud-build': path(`sources/@roots/bud-build/src`),
  '@roots/bud-build/handlers': path(
    `sources/@roots/bud-build/src/handlers`,
  ),
  '@roots/bud-cache': path(`sources/@roots/bud-cache/src`),
  '@roots/bud-compiler': path(`sources/@roots/bud-compiler/src`),
  '@roots/bud-compress': path(`sources/@roots/bud-compress/src`),
  '@roots/bud-dashboard': path(`sources/@roots/bud-dashboard/src`),
  '@roots/bud-extensions': path(`sources/@roots/bud-extensions/src`),
  '@roots/bud-framework': path(`sources/@roots/bud-framework/src`),
  '@roots/bud-framework/extension': path(
    `sources/@roots/bud-framework/src/extension`,
  ),
  '@roots/bud-framework/extension/decorators': path(
    `sources/@roots/bud-framework/src/extension/decorators`,
  ),
  '@roots/bud-hooks': path(`sources/@roots/bud-hooks/src`),
  '@roots/bud-preset-react': path(`sources/@roots/bud-preset-react/src`),
  '@roots/bud-preset-recommend': path(
    `sources/@roots/bud-preset-recommend/src`,
  ),
  '@roots/bud-react': path(`sources/@roots/bud-react/src`),
  '@roots/bud-sass': path(`sources/@roots/bud-sass/src`),
  '@roots/bud-server': path(`sources/@roots/bud-server/src`),
  '@roots/bud-server/middleware': path(
    `sources/@roots/bud-server/src/middleware`,
  ),
  '@roots/bud-support': path(`sources/@roots/bud-support/src`),
  '@roots/bud-swc': path(`sources/@roots/bud-swc/src`),
  '@roots/bud-tailwindcss-theme-json': path(
    `sources/@roots/bud-tailwindcss-theme-json/src`,
  ),
  '@roots/bud-wordpress-externals': path(
    `sources/@roots/bud-wordpress-externals/src`,
  ),
  '@roots/bud/services': path(`sources/@roots/bud/src/services`),
  '@roots/container': path(`sources/@roots/container/src`),
  '@roots/entrypoints-webpack-plugin': path(
    `sources/@roots/entrypoints-webpack-plugin/src`,
  ),
  '@roots/sage': path(`sources/@roots/sage/src`),
  '@roots/wordpress-dependencies-webpack-plugin': path(
    `sources/@roots/wordpress-dependencies-webpack-plugin/src`,
  ),
  '@roots/wordpress-externals-webpack-plugin': path(
    `sources/@roots/wordpress-externals-webpack-plugin/src`,
  ),
  '@roots/wordpress-hmr': path(`sources/@roots/wordpress-hmr/src`),
  '@roots/wordpress-hmr/loader': path(
    `sources/@roots/wordpress-hmr/src/loader`,
  ),
}
