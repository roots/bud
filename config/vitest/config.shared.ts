import type {UserConfig} from 'vitest'

import {env} from 'node:process'

import GithubActionsReporter from 'vitest-github-actions-reporter'

export default {
  hookTimeout: 240000,
  reporters: env.GITHUB_ACTIONS
    ? [`basic`, new GithubActionsReporter()]
    : [`basic`],
  testTimeout: 240000,
  watch: false,
} satisfies UserConfig
