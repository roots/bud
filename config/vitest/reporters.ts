import {env} from 'node:process'

import GithubActionsReporter from 'vitest-github-actions-reporter'

const reporters: Array<any>  = [`basic`]
if (env.GITHUB_ACTIONS) reporters.push(new GithubActionsReporter())

export default reporters
