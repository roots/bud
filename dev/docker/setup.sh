#!/bin/bash

cd /bud

npx npm-cli-login -u test -p test -e test@test.com -r http://verdaccio:4873

yarn config set npmRegistryServer http://verdaccio:4873
yarn config set npmAuthIdent 'test:test'
yarn config set unsafeHttpWhitelist --json '["verdaccio"]'

echo Install
yarn

echo Build
yarn @bud build

echo Done!
