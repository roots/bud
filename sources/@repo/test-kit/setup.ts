import {type Options, default as Project} from './integration-test.js'

export default (options: Options) => new Project(options)

export {type Options, type Project}
