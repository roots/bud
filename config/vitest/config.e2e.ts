import {env} from 'node:process'

import GithubActionsReporter from 'vitest-github-actions-reporter'
import {defineConfig} from 'vitest/config'

import alias from './alias.js'

export default defineConfig({
  test: {
    alias,
    hookTimeout: 240000,
    include: [`tests/e2e/*.test.ts`],
    reporters: [
      `basic`,
      env.GITHUB_ACTIONS ? new GithubActionsReporter() : ``,
    ],
    testTimeout: 240000,
    threads: false,
  },
})
