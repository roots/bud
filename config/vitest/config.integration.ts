import {defineConfig} from 'vitest/config'

import shared from './config.shared'

export default defineConfig({
  test: {
    ...shared,
    include: [`tests/integration/**/*.test.ts`],
    pool: `forks`,
    poolOptions: {
      forks: {
        maxForks: 1,
        minForks: 1,
      },
    },
    setupFiles: [`./config/vitest/setup.integration.ts`],
    watch: false,
  },
})
