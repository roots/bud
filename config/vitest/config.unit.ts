import {env} from 'process'

import {defineConfig} from 'vitest/config'
import GithubActionsReporter from 'vitest-github-actions-reporter'

import alias from './alias.js'

export default defineConfig({
  test: {
    alias,
    coverage: {
      include: [`sources/@roots/*/src/**/*.{ts,tsx}`],
      provider: `v8`,
      reporter: [`text`, `html`],
    },
    exclude: [`sources/@repo/**/*`, `**/node_modules/**/*`],
    hookTimeout: 60000,
    include: [
      `sources/@roots/*/src/*.test.{ts,tsx}`,
      `sources/@roots/*/src/**/*.test.{ts,tsx}`,
      `sources/@roots/*/test/*.test.{ts,tsx}`,
      `sources/@roots/*/test/**/*.test.{ts,tsx}`,
      `sources/@roots/*/tests/*.test.{ts,tsx}`,
      `sources/@roots/*/tests/**/*.test.{ts,tsx}`,
      `tests/unit/**/*.test.ts`,
      `tests/reproductions/**/*.test.ts`,
      `tests/integration/*.test.ts`,
    ],
    includeSource: [`sources/@roots/*/src/**/*.{ts,tsx}`],
    reporters: [
      `default`,
      env.GITHUB_ACTIONS ? new GithubActionsReporter() : ``,
    ],
    testTimeout: 60000,
  },
})
