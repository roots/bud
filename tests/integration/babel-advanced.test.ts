import setup from '@repo/test-kit/setup'
import {beforeAll, describe, expect, it} from 'vitest'

describe(`examples/babel-advanced`, () => {
  const test = setup({label: `@examples/babel-advanced`})

  beforeAll(async () => {
    await test.install()
    await test.build()
  })

  it(`should have expected stdout`, async () => {
    expect(
      test.stdout.split(`\n`).slice(2, -3).join(`\n`),
    ).toMatchSnapshot()
  })

  it(`should have expected manifest.json`, async () => {
    expect(test.manifest).toMatchSnapshot()
  })

  it(`should have expected entrypoints.json`, async () => {
    expect(test.entrypoints).toMatchSnapshot()
  })

  it(`should emit app.js`, async () => {
    expect(test.getAsset(`app.js`)).toMatchSnapshot()
  })

  it(`should emit app.css`, async () => {
    expect(test.getAsset(`app.css`)).toMatchSnapshot()
  })

  it(`should emit runtime.js`, async () => {
    expect(test.getAsset(`runtime.js`)).toMatchSnapshot()
  })
})
