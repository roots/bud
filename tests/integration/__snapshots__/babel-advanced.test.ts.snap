// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`examples/babel-advanced > should emit app.css 1`] = `"body,html{margin:0;padding:0}#root{align-items:center;background:#5813d5;color:#fff;display:flex;font-family:sans-serif;height:100vh;justify-content:center;letter-spacing:.2em;text-align:center;text-transform:uppercase;width:100vw}"`;

exports[`examples/babel-advanced > should emit app.js 1`] = `""use strict";(self.webpackChunk_roots_bud=self.webpackChunk_roots_bud||[]).push([[524],{"./app.js":()=>{var s;null===(s=document.querySelector("#root"))||void 0===s||s.classList.add("init")}},s=>{var e;e="./app.js",s(s.s=e)}]);"`;

exports[`examples/babel-advanced > should emit runtime.js 1`] = `"(()=>{"use strict";var r,e={},o={};function t(r){var n=o[r];if(void 0!==n)return n.exports;var s=o[r]={exports:{}};return e[r](s,s.exports,t),s.exports}t.m=e,r=[],t.O=(e,o,n,s)=>{if(!o){var a=1/0;for(u=0;u<r.length;u++){for(var[o,n,s]=r[u],i=!0,f=0;f<o.length;f++)(!1&s||a>=s)&&Object.keys(t.O).every((r=>t.O[r](o[f])))?o.splice(f--,1):(i=!1,s<a&&(a=s));if(i){r.splice(u--,1);var l=n();void 0!==l&&(e=l)}}return e}s=s||0;for(var u=r.length;u>0&&r[u-1][2]>s;u--)r[u]=r[u-1];r[u]=[o,n,s]},t.o=(r,e)=>Object.prototype.hasOwnProperty.call(r,e),(()=>{var r={121:0};t.O.j=e=>0===r[e];var e=(e,o)=>{var n,s,[a,i,f]=o,l=0;if(a.some((e=>0!==r[e]))){for(n in i)t.o(i,n)&&(t.m[n]=i[n]);if(f)var u=f(t)}for(e&&e(o);l<a.length;l++)s=a[l],t.o(r,s)&&r[s]&&r[s][0](),r[s]=0;return t.O(u)},o=self.webpackChunk_roots_bud=self.webpackChunk_roots_bud||[];o.forEach(e.bind(null,0)),o.push=e.bind(null,o.push.bind(o))})()})();"`;

exports[`examples/babel-advanced > should have expected entrypoints.json 1`] = `
{
  "app": {
    "css": [
      "css/app.css",
    ],
    "js": [
      "js/runtime.js",
      "js/app.js",
    ],
  },
}
`;

exports[`examples/babel-advanced > should have expected manifest.json 1`] = `
{
  "app.css": "css/app.css",
  "app.js": "js/app.js",
  "entrypoints.json": "entrypoints.json",
  "runtime.js": "js/runtime.js",
}
`;

exports[`examples/babel-advanced > should have expected stdout 1`] = `
"│
│ app
│  ◉ js/runtime.js                                                     ✔ 904 bytes
│  ◉ css/app.css                                                       ✔ 231 bytes
│  ◉ js/app.js                                                         ✔ 227 bytes"
`;
